spaceship/lib/spaceship/tunes/build_train.rb

```diff
@@ -66,14 +69,37 @@ def setup
           Tunes::Build.factory(attrs)
         end
 
+        @invalid_builds = @builds.select do |build|
+          build.processing_state == 'processingFailed' || build.processing_state == 'invalidBinary'
+        end
+
+        # This step may not be necessary anymore - it seems as if every processing build will be caught by the
+        # @builds.each below, but not every processing build makes it to buildsInProcessing, so this is redundant
         @processing_builds = (self.raw_data['buildsInProcessing'] || []).collect do |attrs|
           attrs[:build_train] = self
           Tunes::Build.factory(attrs)
         end
 
         # since buildsInProcessing appears empty, fallback to also including processing state from @builds
         @builds.each do |build|
-          @processing_builds << build if build.processing == true && build.valid == true
+          # What combination of attributes constitutes which state is pretty complicated. The table below summarizes
+          # what I've observed, but there's no reason to believe there aren't more states I just haven't seen yet.
+          # The column headers are qualitative states of a given build, and the first column is the observed attributes
+          # of that build.
+          # NOTE: Some of the builds in the build_trains.json fixture do not follow these rules. I don't know if that is
+          # because those examples are older, and the iTC API has changed, or if their format is still a possibility.
+          # The second part of the OR clause in the line below exists so that those suspicious examples continue to be
+          # accepted for unit tests.
+          # +---------------------+-------------------+-------------------+-----------------+--------------------+---------+
+          # |                     | just after upload | normal processing | invalid binary  | processing failed  | success |
+          # +---------------------+-------------------+-------------------+-----------------+--------------------+---------+
+          # |  build.processing = | true              | true              | true            | true               | false   |
+          # |       build.valid = | false             | true              | false           | true               | true    |
+          # | .processing_state = | "processing"      | "processing"      | "invalidBinary" | "processingFailed" | nil     |
+          # +---------------------+-------------------+-------------------+-----------------+--------------------+---------+
+          if build.processing_state == 'processing' || (build.processing && build.processing_state != 'invalidBinary' && build.processing_state != 'processingFailed')
```

Comment:
@KrauseFx As I note in the comment above, everything after the `||` is included just because the pre-existing test cases will otherwise fail. Based on my admittedly limited testing, though, just checking that `processing_state == 'processing` should be enough. I'd want other people to confirm my findings before removing/editing those old test cases and simplifying this conditional.